### PR TITLE
Circle and rectangle

### PR DESCRIPTION
**Pull request by Arens404**

### CODE REVIEW

File: ControlPanel.java [hunk 1]
```diff
@@ -4,34 +4,46 @@ public class ControlPanel {
     static Scanner userInputScanner = new Scanner(System.in);
     static Circle mainCircle = new Circle();
     static Rectangle mainRectangle = new Rectangle();
-    public void controlPanelForOperations(int selectInputNumber){
-        switch(selectInputNumber){
-            case 1:
+    public static void controlPanelForOperations(int selectInputNumber){
```
Comment (Chezo) [inline ControlPanel.java]: still, so much static things in these classes. get rid of them!

File: ControlPanel.java [hunk 2]
```diff
@@ -4,34 +4,46 @@ public class ControlPanel {
...
+        int userInputNumberSelection = Integer.parseInt(userInputScanner.nextLine());
+        switch (userInputNumberSelection) {
+            case 1 -> {
                 double resultOfCirclePerimeter = mainCircle.CirclePerimeter(0);
```
Comment (Chezo) [inline ControlPanel.java]: why is there a magic number?

File: ControlPanel.java [hunk 3]
```diff
@@ -4,34 +4,46 @@ public class ControlPanel {
...
                 System.out.println(resultOfCirclePerimeter);
-                break;
-            case 2:
+            }
+            case 2 -> {
                 double resultOfCircleArea = mainCircle.CircleArea(0);
```
Comment (Chezo) [inline ControlPanel.java]: magic number again

File: ControlPanel.java [hunk 4]
```diff
@@ -4,34 +4,46 @@ public class ControlPanel {
...
+                }
+             }
+            case 4 -> {
                 int resultOfSquarePerimeter = mainRectangle.RectanglePerimeter(0, 0);
                 System.out.println(resultOfSquarePerimeter);
-                break;
-            case 5:
+            }
+            case 5 -> {
                 int resultOfSquareArea = mainRectangle.RectangleArea(0, 0);
```
Comment (Chezo) [inline ControlPanel.java]: magic numbers

File: ControlPanel.java [hunk 5]
```diff
@@ -4,34 +4,46 @@ public class ControlPanel {
...
                 System.out.println(resultOfSquareArea);
-                break;
-            case 6:
+            }
+            case 6 -> {
                 double resultOfRectangleDiagonal = mainRectangle.RectangleDiagonal(0, 0, " ");
```
Comment (Chezo) [inline ControlPanel.java]: magic numbers and empty string parameter?

**Review by Chezo — CHANGES_REQUESTED** [review]

File: Circle.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import java.util.Scanner;
+
+public class Circle extends Main{
+    public double CirclePerimeter(){
+        System.out.print("Enter the circle radius length:");
+        int circleRadius = scanner.nextInt();
+        double circlePerimeter = 2*Math.PI*circleRadius;
```
Comment (Chezo) [inline Circle.java:5]: good variable names finally

File: ControlPanel.java [hunk 7]
```diff
@@ -0,0 +1,64 @@
+import java.util.Scanner;
+
+public class ControlPanel extends Main{
+    public void controlPanelForOperations(int selectInputNumber){
+        Circle mainCircle = new Circle();
+        Rectangle mainRectangle = new Rectangle();
```
Comment (Chezo) [inline ControlPanel.java:5]: why we create these INSTANCES here

File: ControlPanel.java [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+            case 1 -> {
+                double resultOfCirclePerimeter = mainCircle.CirclePerimeter();
+                System.out.println(resultOfCirclePerimeter);
+            }
+            case 2 -> {
+                double resultOfCircleArea = mainCircle.CircleArea();
+                System.out.println(resultOfCircleArea);
+            }
+            case 3 -> {
+                String DegreesOrRadianHelper = " ";
```
Comment (Chezo) [inline ControlPanel.java]: seems like unused variable

File: ControlPanel.java [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+            case 4 -> {
+                int resultOfSquarePerimeter = mainRectangle.RectanglePerimeter();
+                System.out.println(resultOfSquarePerimeter);
+            }
+            case 5 -> {
+                int resultOfSquareArea = mainRectangle.RectangleArea();
+                System.out.println(resultOfSquareArea);
+            }
+            case 6 -> {
+                String DegreesOrRadianHelper = " ";
```
Comment (Chezo) [inline ControlPanel.java]: seems like unused variable

File: Rectangle.java [hunk 10]
```diff
@@ -0,0 +1,51 @@
+import java.util.Scanner;
+
+public class Rectangle extends Main{
```
Comment (Chezo) [inline Rectangle.java]: why its extends main?

File: ControlPanel.java [hunk 11]
```diff
@@ -0,0 +1,64 @@
+import java.util.Scanner;
+
+public class ControlPanel extends Main{
```
Comment (Chezo) [inline ControlPanel.java]: why is extends main?

File: Circle.java [hunk 12]
```diff
@@ -0,0 +1,32 @@
+import java.util.Scanner;
+
+public class Circle extends Main{
```
Comment (Chezo) [inline Circle.java]: why its extends main?

File: Rectangle.java [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+        double RectangleDiagonalOperations = (SideOfaRectangle*SideOfaRectangle) + (AnotherSideOfaRectangle*AnotherSideOfaRectangle);
+        double RectangleSqrt = Math.sqrt(RectangleDiagonalOperations);
+        double RectangleAlphaTangense = RectangleSqrt/SideOfaRectangle;
+        double RectangleShiftTanOperation = rectangleArcCoTangens(RectangleAlphaTangense);
+        switch (DegreesOrRadiant){
+            case "Degrees" -> System.out.print(Math.toDegrees(RectangleShiftTanOperation));
+            case "Radian" -> System.out.print(RectangleShiftTanOperation);
+            default -> System.out.println("That's not an option please select from Degrees/Radiant options!");
+        }
+        return RectangleDiagonalOperations;
```
Comment (Chezo) [inline Rectangle.java]: what kind of operations? its seems like its just a double which is not operation

File: Main.java [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+                    System.out.println(" 6*  Diagonal calculation");
+                    UserInputSelectOptionPanel.controlPanelForOperations(userInputNumber);
+                }
+                case 3 -> {
+                    System.out.println("Shut Down!");
+                    System.exit(0);
+                }
+                default -> System.out.println("This is not an option! Please select form the three option!");
+            }
+        } catch (NumberFormatException IllegalArgumentException){
```
Comment (Chezo) [inline Main.java]: mislead type-name pair

File: Main.java [hunk 15]
```diff
@@ -0,0 +1,36 @@
+import java.util.Scanner;
+public class Main {
+    static Scanner scanner = new Scanner(System.in);
+    public static void main(String[] args) {
+        ControlPanel UserInputSelectOptionPanel = new ControlPanel();
```
Comment (Chezo) [inline Main.java]: should be private

File: Main.java [hunk 16]
```diff
@@ -0,0 +1,36 @@
...
+        Scanner userInputScanner = new Scanner(System.in);
```
Comment (Chezo) [inline Main.java]: should be private and could be final

**Review by Chezo — CHANGES_REQUESTED** [review]

File: Circle.java (hunk 12 above)
Comment (Arens404) [reply Circle.java]: because if we dont, we cant reach the scanner :(

File: ControlPanel.java (hunk 11 above)
Comment (Arens404) [reply ControlPanel.java]: because if we dont, we cant reach the scanner :(

File: Rectangle.java (hunk 10 above)
Comment (Arens404) [reply Rectangle.java]: because if we dont, we cant reach the scanner :(

File: ControlPanel.java (hunk 7 above)
Comment (Arens404) [reply ControlPanel.java:5]: because we control the Rectangle methods and the circle methods here.

Comment (Chezo) [reply ControlPanel.java:5]: its not an answer, there no need to create an instance from the object.

File: ControlPanel.java [hunk 17]
```diff
@@ -0,0 +1,61 @@
+public class ControlPanel{
+    Setter scanner = new Setter();
```
Comment (Chezo) [inline ControlPanel.java:2]: still have multiple scanner what are you hiding in this class, not acceptable solution, use function parameters instead!

File: Main.java (hunk 15 above)
Comment (Chezo) [reply Main.java]: still not private, why u resolved the conversation?

File: Main.java [hunk 18]
```diff
@@ -0,0 +1,36 @@
+
+public class Main {
+    private Setter scanner = new Setter();
+    public static void main(String[] args) {
+        Main main = new Main();
```
Comment (Chezo) [inline Main.java]: why we have a main instance there? big wtf...

File: Main.java [hunk 19]
```diff
@@ -0,0 +1,36 @@
...
+        System.out.println("2 - Rectangle:");
+        System.out.println("3 - Exit:");
+        try{
+            int userInputNumber = main.scanner.input();
+            switch (userInputNumber){
+                case 1 -> {
+                    System.out.println(" 1*  Perimeter calculation");
+                    System.out.println(" 2*  Area calculation");
+                    System.out.println(" 3*  Arc calculation");
+                    UserInputSelectOptionPanel.controlPanelForOperations(userInputNumber);
```
Comment (Chezo) [inline Main.java]: its not meaningful at all, its just too complex to be clear

File: Rectangle.java [hunk 20]
```diff
@@ -0,0 +1,49 @@
+public class Rectangle{
+    private Setter scanner = new Setter();
```
Comment (Chezo) [inline Rectangle.java]: still multiple scanners, use function parameters instead!

**Review by Chezo — CHANGES_REQUESTED** [review]

**Review by Chezo — CHANGES_REQUESTED** [review]
eliminate the scanner specific classes and use inheritance

**Comment by Arens404** [comment]
> eliminate the scanner specific classes and use inheritance

fcd9c574

**Pull request merged**